numpy/f2py/crackfortran.py

```diff
@@ -2481,11 +2504,30 @@ def get_parameters(vars, global_params={}):
                     # FIXME, unused l looks like potential bug
                     l = markoutercomma(v[1:-1]).split('@,@')
 
-            try:
-                params[n] = eval(v, g_params, params)
-            except Exception as msg:
-                params[n] = v
-                outmess('get_parameters: got "%s" on %s\n' % (msg, repr(v)))
+            # Identifying array parameters
+            param_is_array = False
+            for a in vars[n]['attrspec']:
+                if a[:9] == 'dimension':
+                    param_is_array = True
+
+            if param_is_array:
+                v = v.lstrip('(/').rstrip('/)').split(',')
+                # TODO Probably not the most efficient way...
+                v_eval = []
+                for item in v:
+                    try:
+                        v_eval.append(eval(item, g_params, params))
+                    except Exception as msg:
+                        v_eval.append(item)
+                        outmess('get_parameters: got "%s" on %s\n' % (msg, repr(v)))
+                params[n] = v_eval.copy()
```

Comment:
```suggestion
                params[n] = v_eval
```
No need to make a copy of a temporary you never use again